aea/skills/base.py

```diff
@@ -389,18 +389,70 @@ def parse_module(
 class Task(SkillComponent):
     """This class implements an abstract task."""
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        skill_context: Optional[SkillContext] = None,
+        **kwargs
+    ):
         """
         Initialize a task.
 
-        :param skill_context: the skill context
+        :param name: the name of the task (mandatory)
+        :param skill_context: the skill context (mandatory)
         :param kwargs: keyword arguments.
         """
-        super().__init__(**kwargs)
-        self.completed = False
+        super().__init__(name=name, skill_context=skill_context, **kwargs)
+
+        self._executed = False
+        # this is where we store the result.
+        self._result = None
```

Comment:
`result` stores the computed result, if any -> accessed through the associated property. Raises error if the task has not been executed yet.

Reply:
This should be documented too